ggshield/pypi.py

```diff
@@ -0,0 +1,107 @@
+import re
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Dict, List, Set
+
+import click
+
+from ggshield.config import Config
+from ggshield.output import OutputHandler
+from ggshield.path import get_files_from_paths
+from ggshield.scan import Files, Result, ScanCollection
+from ggshield.utils import SupportedScanMode
+
+
+PYPI_DOWNLOAD_TIMEOUT = 30
+
+
+def save_package_to_tmp(temp_dir: str, package_name: str) -> None:
+    command: List[str] = [
+        "pip",
+        "download",
+        package_name,
+        "--dest",
+        temp_dir,
+        "--no-deps",
+    ]
+
+    try:
+        click.echo("Downloading pip package... ", nl=False)
+        subprocess.run(
+            command,
+            check=True,
+            stderr=subprocess.PIPE,
+            timeout=PYPI_DOWNLOAD_TIMEOUT,
+        )
+        click.echo("OK")
+
+    except subprocess.CalledProcessError:
+        raise click.ClickException(f'Failed to download "{package_name}"')
+
+    except subprocess.TimeoutExpired:
+        raise click.ClickException('Command "{}" timed out'.format(" ".join(command)))
+
+
+def get_files_from_package(
+    temp_dir: str,
```

Comment:
nit: rename `temp_dir` to more general `package_dir`, in the function's context the directory doesn't have to be temporary